### ISSUE TITLE
[Bug]: Xenos might CreateRemoteThread without admin privilege, fucking up YimMenu OS permissions. Add a log to user when that happens

### ISSUE THREAD

**Issue by Deadlineem** [Yimura/YimMenu#3499]
### Describe the bug

This issue seems to be for some windows 10 users, its not allowing the scripts_config folder or files to be read, which is causing the lua scripts to fail to load if they cant read/write to the config file.  In this case HSConfig.lua

https://github.com/YimMenu-Lua/Harmless-Scripts/issues/4

### Steps To Reproduce

If you are on windows 11, this issue does not appear as windows 11 has 3 settings for readonly
Enabled/disabled/passive 

Passive shows up as a - but this is not part of windows 10 systems, win10 and below only have enable/disable

Setting readonly to enabled and running yim will reproduce the error in the link provided on windows 11.

### Expected Behavior

The folder should not be readonly by default.

### Logs

In provided link

### Store

Other (please mention in "Additional context")

### Language

English

### YimMenu Language

English (US)

### Additional context

OS windows 10 and under

### Build

- [X] I've confirmed the issue exists on the latest version of YimMenu

### Coload

- [X] I've tested this without co-loading anything.

**Comment by xiaoxiao921** [comment]
Windows read only attribute flag on folders is initially in an indeterminate state, if you had read the log correctly of the issue you linked you should've seen that there is already similar problems about permissions for this user (failing to read regedit keys and such properly), this is just an issue with their antivirus or similar blocking the OS handles from the YimMenu.dll to function properly

**Comment by Deadlineem** [comment]
> Windows read only attribute flag on folders is initially in an indeterminate state, if you had read the log correctly of the issue you linked you should've seen that there is already similar problems about permissions for this user (failing to read regedit keys and such properly), this is just an issue with their antivirus or similar blocking the OS handles from the YimMenu.dll to function properly

So the only solution is to manually disable the readonly or disable their virus protection?  I guess that makes sense but this should be noted/documented on the readme or docs.  Some people have some weird virus scanners though which could be a major factor as you said.

**Comment by xiaoxiao921** [comment]
The read only flag has nothing to do with the user problem. We don't create folders with read only enabled. It's just something that is installed from their PC that is blocking WinAPI handles of YimMenu.

**Comment by Deadlineem** [comment]
> The read only flag has nothing to do with the user problem. We don't create folders with read only enabled. It's just something that is installed from their PC that is blocking WinAPI handles of YimMenu.

I wonder if the two users having this issue have a common antivirus software or maybe not having the proper user account privileges or having UAC disabled/enabled.  Idk what the cause is specifically but ill try and find out.

There are 2 people with the issue on that post.

**Comment by xiaoxiao921** [comment]
I'm gonna reopen this and use this issue as a Todo for us that we should show message to the user that xenos should be launched as admin and that no antivirus should block the YimMenu dll

**Comment by FIying-Scotsman** [comment]
> I'm gonna reopen this and use this issue as a Todo for us that we should show message to the user that xenos should be launched as admin and that no antivirus should block the YimMenu dll

Would just want to add to the message - **_users should not outright disable their antivirus. Add Yimmenu and Xenos as exceptions._**

There's a infostealer campaign going about which is abusing this to compromise victim PCs to steal their accounts to sell on their forum. The infostealer is packed in fake game cheats including injectors and drivers and prompts the user to disable their AV to function (as soon as they click on it the infostealer is loaded regardless but it'll wait until the AV is disabled before running fully). 

It's incredibly bad practice to just disable your AV regardless.

**Comment by gir489returns** [comment]
I don't think we should log this. We clearly specify how the binary should be loaded into memory, and it should always be done via LoadLibraryA so we can resolve the DLL properly. Manually Mapping the DLL like it's 2003 isn't going to stop anything. Rockstar's anti-cheat manually walks the PEB space, what's even the point? But you probably have no idea what I'm talking about, you probably watched some stupid YouTube guide with skibiti toilet playing in the background along with Minecraft parkour and Suibway surfers to keep your attention, and some 13-year-old with a heavy Indian accent told you to use Xenos.

**Comment by xiaoxiao921** [comment]
This is planned, actually, we finally found the issue on why the rpf stuff is failing etc, just logging that the thread was made with the right privilege or not. This bug happens when xenos default injection method is used, it uses CreateRemoteThread and don't have the right thread privilege if not launched as admin

**Comment by gir489returns** [comment]
> This bug happens when xenos default injection method is used, it uses CreateRemoteThread and don't have the right thread privilege if not launched as admin

No? 

https://github.com/DarthTon/Blackbone/blob/5ede6ce50cd8ad34178bfa6cae05768ff6b3859b/src/BlackBone/Process/ProcessModules.cpp#L484-L493

Xenos uses Blackbone. Blackbone's **ACTUAL** default injection just finds a random thread and injects code into it and makes it call LoadLibraryW (Which is the WCHAT_T version of LoadLibraryA). I'm like 99% certain these people are using the manual map mode of Xenos.
